Sources/Articles/Documentation.docc/2024/swiftpm-snippets/SwiftPM Snippets.md

```diff
@@ -0,0 +1,264 @@
+```bash
+$ swift build --target SnippetsExample_I
+```
+
+You can run a Snippet with the `swift run` command.
+
+```bash
+$ swift run SnippetsExample_I
+```
+
+```text
+Building for debugging...
+[5/5] Linking SnippetsExample_I
+Build complete! (0.97s)
+Hi Barbie!
+```
+
+You could also run a Snippet with release optimizations, just like any other executable target.
+
+```bash
+$ swift run -c release SnippetsExample_I
+```
+
+```text
+Building for production...
+[17/17] Linking SnippetsExample_I
+Build complete! (9.73s)
+Hi Barbie!
+```
+
+>   Note:
+>   If the Snippet name contains special characters, you should pass the file name as-is to the `--target` flag. For example, to run a Snippet named `Foo bar.swift`, you would use `swift run 'Foo bar'` and not `Foo_bar`.
+
+
+## Embedding Snippets in documentation
+
+Most modern documentation engines support embedding Snippets in Markdown documentation via the `@Snippet` block directive.
+
+Let’s create a documentation bundle for the `SnippetsExample` target.
+
+```bash
+$ mkdir -p Sources/SnippetsExample/docs.docc
+```
+
+>   Tip:
+>   You can name the documentation bundle anything you like, as long as it has a `.docc` extension. The name comes from the DocC tool, but most other documentation engines will recognize it too.
+
+Create a markdown article named `My article.md` in the `docs.docc` directory.
+
+@Code(name: "My article.md", file: "My article (2).md.txt")
+
+In the example above, we have specified the Snippet to include by `path` identity. Despite its naming, the `path` syntax is not a file path. The first component is the name of the package **as specified by the ``PackageDescription/Package/name`` field in the manifest**. The second component is always the string `Snippets`. The third component is the Snippet ID, which is the name of the Snippet file without the `.swift` extension. If the Snippet ID contains special characters, you should pass the ID as-is, without replacing any characters.
+
+Some documentation engines such as Unidoc support referencing Snippets by `id`.
+
+@Code(name: "My article.md", file: "My article (1).md.txt")
+
+>   Important:
+>   DocC does not currently support referencing Snippets by ID. Instead, you must use the fully-qualified `path` syntax to reference a Snippet.
+
+You should now have a project layout that looks like this:
+
+```
+📂 swift-snippets
+├── 📂 Snippets
+│   └── 📄 SnippetsExample_I.swift
+├── 📂 Sources
+│   └── 📂 SnippetsExample
+│       ├── 📂 docs.docc
+│       │   └── 📄 My article.md
+│       └── 📄 anchor.swift
+├── 📄 Package.swift
+└── 📄 .gitignore
+```
+
+
+### Previewing Snippets with DocC
+
+Many developers find [DocC](https://github.com/apple/swift-docc) helpful for previewing documentation locally. To use DocC, add the [swift-docc-plugin](https://github.com/apple/swift-docc-plugin) to the package manifest.
+
+@Code(name: Package.swift, file: Manifest.3.swift)
+
+You can then launch DocC with the `preview-documentation` package subcommand:
```

Comment:
The documentation preview can be launched using the `preview-documentation` subcommand.

We should also mention why we need the --disable-sandbox flag.

Also we have to create an empty .swift file under the SnippetsExample target, otherwise the package won't compile, since SPM can't find any sources.

Reply:
added a note about empty file earlier in the tutorial, but i personally have no idea why `--disable-sandbox` is needed, i have cargo culting that for a long time. the [official DocC README](https://github.com/swiftlang/swift-docc-plugin?tab=readme-ov-file#previewing-documentation) doesn’t mention why it is needed.